src/MAVLink/StatusTextHandler.cc

```diff
@@ -139,7 +139,7 @@ void StatusTextHandler::resetErrorLevelMessages()
 
 void StatusTextHandler::handleTextMessage(MAV_COMPONENT compId, MAV_SEVERITY severity, const QString &text, const QString &description)
 {
-    QString htmlText(text);
+    QString htmlText(text.toHtmlEscaped());
```

Comment:
The problem with escaping here is that it will be escaped twice in case of the events code path from https://github.com/mavlink/qgroundcontrol/blob/master/src/Vehicle/Vehicle.cc#L1230

That's why I originally did it outside, but did not consider that calibration was affected.
The simplest would be to add a flag as argument to indicate if it's escaped already. But I do agree, would be good to simplify this and avoid the back and forth.

Reply:
My understanding from code inspection are these path(s):
* StatusTextHandler::mavlinkMessageReceived is the only thing which sucks up status text messages from mavlink
  * It eventually emits StatusTextHandler::textMessageReceived
* Vehicle catches StatusTextHandler::textMessageReceived and calls Vehicle::_textMessageReceived which...
  * Signals Vehicle::textMessageReceived
  * Calls StatusTextHandler::handleTextMessage which...
    * emits newErrorMessage which is escaped text
    * Stuffs escaped text into the message list
* Events come in through Vehicle::_handleEvent - not sure how that gets called. Which then...
  * Also calls StatusTextHandler::handleTextMessage

I don't see how you end up going through StatusTextHandler::handleTextMessage twice. I also don't see how to test this since I don't know how to generate an event.

Reply:
Escaping for the events is done here: https://github.com/mavlink/qgroundcontrol/blob/be1e2d46c95812d3ad700bdc4807eca95829be6b/src/MAVLink/LibEvents/EventHandler.cc#L48.
This is because they can contain URL's which should not be escaped.

If you connect to PX4 SITL and do a takeoff you should get some events.

Reply:
Ah, ok. I get it now.

Reply:
@bkueng I think I have this correct now. Can you take a look again?

* Status text coming in from events is known to already be escaped
* Regular textMessageReceived signal is not escaped
* StatusTextHandler::handleTextMessage -> StatusTextHandler::handleHTMLEscapedText name change which assumes escaped text on the way in
* Raw status text from vehicle is escaped before call to StatusTextHandler::handleHTMLEscapedText

I haven't had time to test yet. Take me a bit to get a SITL setup to work again on OSX arm.